### PR TITLE
drive math v3

### PR DESCRIPTION
**Pull request by HarryWJackson1996**
Drive maths version 3. Added Drive.h and Drive.cpp

Fixed git user problems.

### CODE REVIEW

**Review by SamanthaDawsonBanks — APPROVED** [review]
other than the previous git issues, this code is good.
as discussed there does need to be some refracting so that the maths is in a math function(or class) that is then called by drive etc etc but that will probably happen as part of the upcoming interface re-factor

File: mcomp_C/common/movement/Drive.cpp
Comment (stevempope) [inline mcomp_C/common/movement/Drive.cpp:14]: The Math library comes with an very precise version of Pi, so I would remove this line and replace the later use of PI with Math.Pi

File: mcomp_C/common/movement/Drive.cpp
Comment (stevempope) [inline mcomp_C/common/movement/Drive.cpp:16]: This signature should be Waypoint Drive (Waypoint target)?

Returns a Waypoint, name is drive, handed a waypoint called target. Anywhere we use x and y from target we just call target.getX or .getY rather than use more storage on the Arduino. We have more "space" on the Mega than the Uno but it's still good practice to use the smallest space possible.

File: mcomp_C/common/movement/Drive.cpp
Comment (stevempope) [inline mcomp_C/common/movement/Drive.cpp:21]: See comment on line 16.

File: mcomp_C/common/movement/Drive.cpp
Comment (stevempope) [inline mcomp_C/common/movement/Drive.cpp:27]: See comment on line 16.

File: mcomp_C/common/movement/Drive.cpp
Comment (stevempope) [inline mcomp_C/common/movement/Drive.cpp:31]: See comment on line 16.

File: mcomp_C/common/movement/Drive.cpp
Comment (stevempope) [inline mcomp_C/common/movement/Drive.cpp:35]: Phi is a constant?

File: mcomp_C/common/movement/Drive.h
Comment (stevempope) [inline mcomp_C/common/movement/Drive.h:27]: This should take a Waypoint, rather than an x and y

**Review by stevempope — COMMENTED** [review]
Overall I am happy with the code. There are some signature bits to sort to make the code more efficient.

We should be waiting for two reviews of the code before we merge, as it is now very difficult to "review".

Other than this all good to extend.

File: mcomp_C/common/movement/Drive.cpp
Comment (SamanthaDawsonBanks) [inline mcomp_C/common/movement/Drive.cpp:20]: [for v4] there is no need for robot cords as the WP will always be relative to the bot

File: mcomp_C/common/movement/Drive.cpp
Comment (SamanthaDawsonBanks) [inline mcomp_C/common/movement/Drive.cpp:41]: [for v4] why is this one 0?

File: mcomp_C/common/movement/Drive.h
Comment (SamanthaDawsonBanks) [inline mcomp_C/common/movement/Drive.h:11]: [for v4]  will need a return WP

**Review by SamanthaDawsonBanks — COMMENTED** [review]
see comments [for v4]

**Pull request merged**
